core/common/src/main/java/alluxio/conf/ReconfigurableRegistry.java

```diff
@@ -45,12 +48,24 @@ public static synchronized boolean unregister(Reconfigurable listener) {
    * @return false if no listener related to the given property, otherwise, return false
    */
   public static synchronized boolean update() {
-    for (Reconfigurable listener : new LinkedList<>(LISTENER_LIST)) {
+    for (Reconfigurable listener : LISTENER_LIST) {
       listener.update();
     }
     return true;
   }
 
   // prevent instantiation
   private ReconfigurableRegistry() {}
+
+  /**
+   * When the property was reconfigured, this function will be invoked.
+   * This property listeners will be notified.
+   *
+   * @param changedProperties the changed properties
+   */
+  public static synchronized void update(Map<PropertyKey, Object> changedProperties) {
+    for (Reconfigurable listener : LISTENER_LIST) {
+      listener.update(changedProperties);
```

Comment:
try-catch around each listener update so that if one listener throws an exception, it won't break the for loop, and skip the rest of the listeners.